[2025-04-24T20:58:16.534+0000] {taskinstance.py:1956} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: raw_prices_data_downloader_branching.branch_TATASTEEL.NS manual__2025-04-24T20:58:15.012508+00:00 [queued]>
[2025-04-24T20:58:16.541+0000] {taskinstance.py:1956} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: raw_prices_data_downloader_branching.branch_TATASTEEL.NS manual__2025-04-24T20:58:15.012508+00:00 [queued]>
[2025-04-24T20:58:16.541+0000] {taskinstance.py:2170} INFO - Starting attempt 1 of 2
[2025-04-24T20:58:16.556+0000] {taskinstance.py:2191} INFO - Executing <Task(BranchPythonOperator): branch_TATASTEEL.NS> on 2025-04-24 20:58:15.012508+00:00
[2025-04-24T20:58:16.564+0000] {standard_task_runner.py:60} INFO - Started process 3266 to run task
[2025-04-24T20:58:16.569+0000] {standard_task_runner.py:87} INFO - Running: ['***', 'tasks', 'run', 'raw_prices_data_downloader_branching', 'branch_TATASTEEL.NS', 'manual__2025-04-24T20:58:15.012508+00:00', '--job-id', '205', '--raw', '--subdir', 'DAGS_FOLDER/raw_price_data_testing.py', '--cfg-path', '/tmp/tmpbokli9lw']
[2025-04-24T20:58:16.582+0000] {standard_task_runner.py:88} INFO - Job 205: Subtask branch_TATASTEEL.NS
[2025-04-24T20:58:16.722+0000] {task_command.py:423} INFO - Running <TaskInstance: raw_prices_data_downloader_branching.branch_TATASTEEL.NS manual__2025-04-24T20:58:15.012508+00:00 [running]> on host 8291240c5d79
[2025-04-24T20:58:16.859+0000] {taskinstance.py:2480} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='raw_prices_data_downloader_branching' AIRFLOW_CTX_TASK_ID='branch_TATASTEEL.NS' AIRFLOW_CTX_EXECUTION_DATE='2025-04-24T20:58:15.012508+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-04-24T20:58:15.012508+00:00'
[2025-04-24T20:58:16.874+0000] {logging_mixin.py:188} WARNING - /home/***/.local/lib/python3.10/site-packages/pymongo/synchronous/mongo_client.py:2709 UserWarning: MongoClient opened before fork. May not be entirely fork-safe, proceed with caution. See PyMongo's documentation for details: https://dochub.mongodb.org/core/pymongo-fork-deadlock
[2025-04-24T20:58:16.910+0000] {taskinstance.py:2698} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 433, in _execute_task
    result = execute_callable(context=context, **execute_callable_kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/operators/python.py", line 233, in execute
    return self.do_branch(context, super().execute(context))
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/operators/python.py", line 199, in execute
    return_value = self.execute_callable()
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/operators/python.py", line 216, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
  File "/opt/airflow/dags/raw_price_data_testing.py", line 93, in branch_condition
    if existing_doc['upload_time'].date() == current_date:
TypeError: 'NoneType' object is not subscriptable
[2025-04-24T20:58:16.930+0000] {taskinstance.py:1138} INFO - Marking task as UP_FOR_RETRY. dag_id=raw_prices_data_downloader_branching, task_id=branch_TATASTEEL.NS, execution_date=20250424T205815, start_date=20250424T205816, end_date=20250424T205816
[2025-04-24T20:58:16.947+0000] {standard_task_runner.py:107} ERROR - Failed to execute job 205 for task branch_TATASTEEL.NS ('NoneType' object is not subscriptable; 3266)
[2025-04-24T20:58:16.981+0000] {local_task_job_runner.py:234} INFO - Task exited with return code 1
[2025-04-24T20:58:17.017+0000] {taskinstance.py:3280} INFO - 0 downstream tasks scheduled from follow-on schedule check
